centrifuge.wdl

```diff
@@ -1,124 +1,201 @@
-        ~{true="-2" false="" defined(read2)} ~{sep=',' read2} \
-        ~{true="-U" false="" length(select_first([unpairedReads])) > 0} ~{sep=',' unpairedReads} \
-        ~{"--report-file " + reportFilePath} \
-        ~{"--min-hitlen " + minHitLen} \
-        ~{"--min-totallen " + minTotalLen} \
-        ~{"--met-file " + metFilePath} \
-        ~{true="--host-taxids " false="" defined(hostTaxIds)} ~{sep=',' hostTaxIds} \
-        ~{true="--exclude-taxids " false="" defined(excludeTaxIds)} ~{sep=',' excludeTaxIds} \
-        ~{true="| gzip -c >" false="-S" compressOutput} ~{finalOutputPath}
+        ~{true="-1 " false="-U " defined(read2)} ~{sep="," read1} \
+        ~{"-2 "} ~{sep="," read2} \
+        ~{"-S " + outputPrefix + "/" + outputName + "_classification.tsv"} \
+        ~{"--report-file " + outputPrefix + "/" + outputName + "_output_report.tsv"}
+    }
+
+    output {
+        File outputMetrics = outputPrefix + "/" + outputName + "_alignment_metrics.tsv"
+        File outputClassification = outputPrefix + "/" + outputName + "_classification.tsv"
+        File outputReport = outputPrefix + "/" + outputName + "_output_report.tsv"
+    }
+
+    runtime {
+        cpu: cores
+        memory: memory
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        inputFormat: {description: "The format of the read file(s).", category: "required"}
+        phred64: {description: "If set to true, Phred+64 encoding is used.", category: "required"}
+        minHitLength: {description: "Minimum length of partial hits.", category: "required"}
+        indexPrefix: {description: "The basename of the index for the reference genomes.", category: "required"}
+        read1: {description: "List of files containing mate 1s, or unpaired reads.", category: "required"}
+        outputPrefix: {description: "Output directory path + output file prefix.", category: "required"}
+        outputName: {description: "The base name of the outputPrefix.", category: "required"}
+        read2: {description: "List of files containing mate 2s.", category: "common"}
+        trim5: {description: "Trim <int> bases from 5' (left) end of each read before alignment.", category: "common"}
+        trim3: {description: "Trim <int> bases from 3' (right) end of each read before alignment.", category: "common"}
+        reportMaxDistinct: {description: "It searches for at most <int> distinct, primary assignments for each read or pair.", category: "common"}
+        hostTaxIDs: {description: "A comma-separated list of taxonomic IDs that will be preferred in classification procedure.", category: "common"}
+        excludeTaxIDs: {description: "A comma-separated list of taxonomic IDs that will be excluded in classification procedure.", category: "common"}
+    }
+}
+
+task Inspect {
+    input {
+        String printOption = "fasta"
+        String indexBasename
+        String outputPrefix
+
+        Int? across
+
+        Int cores = 1
```

Comment:
This input isn't getting used.

Reply:
It is by the runtime right? Or can users set that without this input?

Reply:
But it isn't used by the command, so the tool will always just use one thread. There is no point in setting a (configurable) number of  cores in the runtime section.